### PR TITLE
Handle complex types in convert_element_type | fix(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Previously when we call e.g. `convert_element_type(a, dtype=complex64)`, we create a node with `COMPLEX64` output. This is not what we want because we always the real representation of a complex number in the ONNX graphs.

This change updates the logic to handle when the specified dtype is a complex type, in which case we add an additional axis to the end of the tensor and fill it with zeros to represent the imaginary part.

Tested locally:

```
>>> prims_convert_element_type(np.zeros(1), 14)
array([[0., 0.]], dtype=float32)
>>> prims_convert_element_type(np.zeros(1), 15)
array([[0., 0.]])
```

Fix https://github.com/microsoft/onnxscript/issues/1122

### CODE REVIEW

**Review by thiagocrepaldi — COMMENTED** [review]
There is a bucnh of CI failures

```bash
ERROR onnxscript/function_libs/tools/torch_lib/deduce_type_constraints_test.py - ValueError: ERROR: Unsupported expression type <class 'ast.Attribute'>.
at: Function 'prims_convert_element_type', line 5
    if dtype == COMPLEX128.dtype:
                ^
ERROR onnxscript/function_libs/tools/torch_lib/deduce_type_constraints_test.py - ValueError: ERROR: Unsupported expression type <class 'ast.Attribute'>.
at: Function 'prims_convert_element_type', line 5
    if dtype == COMPLEX128.dtype:
                ^
ERROR onnxscript/function_libs/torch_lib/graph_building_test.py - ValueError: ERROR: Unsupported expression type <class 'ast.Attribute'>.
at: Function 'prims_convert_element_type', line 5
    if dtype == COMPLEX128.dtype:
                ^
ERROR onnxscript/function_libs/torch_lib/graph_building_test.py - ValueError: ERROR: Unsupported expression type <class 'ast.Attribute'>.
at: Function 'prims_convert_element_type', line 5
    if dtype == COMPLEX128.dtype:
                ^
ERROR onnxscript/tests/function_libs/torch_lib/ops_test.py - ValueError: ERROR: Unsupported expression type <class 'ast.Attribute'>.
at: Function 'prims_convert_element_type', line 5
    if dtype == COMPLEX128.dtype:
                ^
ERROR onnxscript/tests/function_libs/torch_lib/ops_test.py - ValueError: ERROR: Unsupported expression type <class 'ast.Attribute'>.
at: Function 'prims_convert_element_type', line 5
    if dtype == COMPLEX128.dtype:
                ^
```

Do we need additional testing for the new types we will support?

**Comment by justinchuby** [comment]
Fixed

File: onnxscript/function_libs/torch_lib/ops/prims.py [hunk 1]
```diff
@@ -217,10 +220,28 @@ def prims_conj_physical(self: TensorType) -> TensorType:
 
 
 @torch_op("prims::convert_element_type")
-def prims_convert_element_type(a: TensorType, dtype: int) -> TensorType:
+def prims_convert_element_type(a: TReal, dtype: int) -> TReal:
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/prims.py:223]:
```suggestion
def prims_convert_element_type(a: TReal, dtype: int) -> TensorType:
```

**Comment by titaiwangms** [comment]
Have you tried it with models? Would if/else drag down the performance?

**Comment by justinchuby** [comment]
I have not tested with with models. I will see if I can get my hands on the googlefnet model. If/else will impact performance but we will rely on constant folding to clear them out as we do in other cases for consistency.

**Review by titaiwangms — APPROVED** [review]

**Comment by justinchuby** [comment]
I am going to merge this for now.

**Pull request merged**